src/DataObject/GridColumnConfig/Operator/Substring.php

```diff
@@ -84,13 +91,35 @@ public function getLabeledValue(array|ElementInterface $element): ResultContaine
             if ($isArrayType) {
                 $result->value = $valueArray;
             } else {
-                $result->value = $valueArray[0];
+                $result->value = $valueArray[0] ?? '';
             }
         }
 
         return $result;
     }
 
+    private function convertToString(mixed $value): string
+    {
+        if (is_array($value)) {
+            $output = implode(
+                ' ',
+                array_map(
+                    fn($v) => is_scalar($v) ? (string)$v : '',
```

Comment:
Objects with `__toString` as array values can never happen?

If it is possible:
```suggestion
                    fn($v) => is_scalar($v) || $v instanceof \Stringable ? (string)$v : '',
```

or maybe better recursively call this method on all array values?

Reply:
Thank you for all the feedback, it also crossed my mind of some recursion, but it gets into some area where the data is really out of the scope or usefulness for a substring functionality, also i was not sure if a space separator for implode would be good.